testbed/tests/window/test_window.py

```diff
@@ -138,13 +143,191 @@ async def test_move_and_resize(main_window, main_window_probe, capsys):
         finally:
             main_window.content = orig_content
 
-    async def test_full_screen(main_window, main_window_probe):
-        """Window can be made full screen"""
-        main_window.full_screen = True
-        await main_window_probe.wait_for_window("Full screen is a no-op")
+    @pytest.mark.parametrize(
+        "initial_state, final_state",
+        [
+            # Direct switch from NORMAL:
+            (WindowState.NORMAL, WindowState.FULLSCREEN),
+            (WindowState.NORMAL, WindowState.PRESENTATION),
+            # Direct switch from FULLSCREEN:
+            (WindowState.FULLSCREEN, WindowState.NORMAL),
+            (WindowState.FULLSCREEN, WindowState.PRESENTATION),
+            # Direct switch from PRESENTATION:
+            (WindowState.PRESENTATION, WindowState.NORMAL),
+            (WindowState.PRESENTATION, WindowState.FULLSCREEN),
+        ],
+    )
+    async def test_window_state_direct_change(
```

Comment:
Why differentiate "direct" from "same" state changes? We're looking to validate that moving from X to Y works - for *any* value of X and Y, including X == Y.  In what way are the "same state" tests different?

Reply:
The "same" stage changes test basically check that the reassignment of the window state to the same current state is actually a no-op i.e., there is no OS delay(given that there are no intermediate state assignments in-between).